transport/config/config.go

```diff
@@ -0,0 +1,146 @@
+package config
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/allegro/akubra/metrics"
+)
+
+// ClientTransportDetail properties
+type ClientTransportDetail struct {
+	// MaxIdleConns see: https://golang.org/pkg/net/http/#Transport
+	// Default 0 (no limit)
+	MaxIdleConns int `yaml:"MaxIdleConns" validate:"min=0"`
+	// MaxIdleConnsPerHost see: https://golang.org/pkg/net/http/#Transport
+	// Default 100
+	MaxIdleConnsPerHost int `yaml:"MaxIdleConnsPerHost" validate:"min=0"`
+	// IdleConnTimeout see: https://golang.org/pkg/net/http/#Transport
+	// Default 0 (no limit)
+	IdleConnTimeout metrics.Interval `yaml:"IdleConnTimeout"`
+	// ResponseHeaderTimeout see: https://golang.org/pkg/net/http/#Transport
+	// Default 5s (no limit)
+	ResponseHeaderTimeout metrics.Interval `yaml:"ResponseHeaderTimeout"`
+	// DisableKeepAlives see: https://golang.org/pkg/net/http/#Transport
+	// Default false
+	DisableKeepAlives bool `yaml:"DisableKeepAlives"`
+}
+
+// ClientTransportMatchers properties
+type ClientTransportMatchers struct {
+	Method     string `yaml:"Method" validate:"max=64"`
+	Path       string `yaml:"Path" validate:"max=64"`
+	QueryParam string `yaml:"QueryParam" validate:"max=64"`
+}
+
+// MatchersCompiledRules properties
+type MatchersCompiledRules struct {
+	MethodRegexp     *regexp.Regexp
+	PathRegexp       *regexp.Regexp
+	QueryParamRegexp *regexp.Regexp
+	IsCompiled       bool
+}
+
+// Transport properties
+type Transport struct {
+	Name                  string                  `yaml:"Name"`
+	Matchers              ClientTransportMatchers `yaml:"Matchers"`
+	MatchersCompiledRules MatchersCompiledRules
+	Details               ClientTransportDetail `yaml:"Details"`
+}
+
+// Transports map with Transport
+type Transports []Transport
+
+// compileRule
+func (t *Transport) compileRule(regexpRule string) (compiledRule *regexp.Regexp, err error) {
+	if len(regexpRule) > 0 {
+		compiledRule, err = regexp.Compile(regexpRule)
+	}
+	return
+}
+
+// transportFlags for internal matching func
+type transportFlags struct {
+	declared bool
+	matched  bool
+	empty    bool
+}
+
+// compileRules prepares precompiled regular expressions for rules
+func (t *Transport) compileRules() error {
+	if !t.MatchersCompiledRules.IsCompiled {
```

Comment:
Readability:
```
if t.MatchersCompiledRules.IsCompiled {
    return nil
}
```
And reduce one indentation

Reply:
BTW, is this sentinel really necessary?

Reply:
Yes, only in first iteration (in loop by transports config) we compiling reg. expr. (it's CPU consuming).

Reply:
Ok let it be, but invert this condition so there will be no unnecessary nesting. Also remeber to remove `if len(t.Matchers.Method) > 0 {` checks in this method because it's checked inside `compileRule` method

Reply:
`compileRules` func has own checks for not empty reg. exp. for compiling only filled rules
`matchTransportFlags` func it's simple algo. and we need here the same information
I propose to extract below conditions to private func. `checkEmptyMatchers` if it's still necessary.

Reply:
Ok, so method `compileRule` is useless, because all it does is same pattern length check and single line call to regexCompile...